import-scripts/clear_cbioportal_persistence_cache.sh

```diff
@@ -196,27 +196,42 @@ else
-    echo "warning : the attempt to restart portal '$portal_id' failed : $KUBECTL_BINARY returned status $webapp_restart_status" >&2
-    echo "          nonetheless, a command to clear the persistence cache will now occur." >&2
+# WORKAROUND(12/2/2025): Zain is currently performing a rolling update of the msk cluster.
+# During this time, whenever the persistence cache for an
+# internal portal is cleared, we should clear it for *both* the eks and eksargocd clusters.
+# Once the transition to the eksargocd cluster is complete
+# (and the portal_to_cluster_map above has been updated), this
+# workaround can be removed.
+declare -a kubeconfig_filepaths
+kubeconfig_filepaths+=("$kubeconfig_filepath")
+if [ "$cluster_id" == "$CLUSTER_ID_EKS" ] ; then
+    kubeconfig_filepaths+=("$EKSARGOCD_CLUSTER_KUBECONFIG")
 fi
 
-cache_service_basename="${portal_to_cache_service_basename[$portal_id]}"
-cache_leader_pod_name="${cache_service_basename}-master-0"
-cache_database_number="${portal_to_cache_database_number[$portal_id]}"
-if [ -n "$cache_service_basename" ] && database_number_is_valid "$cache_database_number" ; then
-    "$KUBECTL_BINARY" $kubeconfig_arg exec -t -n "default" "$cache_leader_pod_name" -- "/bin/bash" -c "redis-cli -a \$REDIS_PASSWORD -n $cache_database_number FLUSHDB"
-    cache_flush_status=$?
-    if [ $cache_flush_status -ne 0 ] ; then
-        echo "error encountered during attempt to clear portal persistence cache for portal $portal_id : $KUBECTL_BINARY returned status $cache_flush_status" >&2
-        exit 1
+for kubeconfig_filepath in "${kubeconfig_filepaths[@]}"; do
+    kubeconfig_arg="--kubeconfig=$kubeconfig_filepath"
+
+    "$KUBECTL_BINARY" $kubeconfig_arg set env deployment "$deployment_id" --env="LAST_RESTART=$(date)"
+    webapp_restart_status=$?
+    if [ $webapp_restart_status -ne 0 ] ; then
+        echo "warning : the attempt to restart portal '$portal_id' failed : $KUBECTL_BINARY returned status $webapp_restart_status" >&2
+        echo "          nonetheless, a command to clear the persistence cache will now occur." >&2
     fi
-else
-    echo "portal $portal_id has no persistence cache service or has no valid database number. skipping."
-    exit 0
-fi
+
+    cache_service_basename="${portal_to_cache_service_basename[$portal_id]}"
+    cache_leader_pod_name="${cache_service_basename}-master-0"
+    cache_database_number="${portal_to_cache_database_number[$portal_id]}"
+    if [ -n "$cache_service_basename" ] && database_number_is_valid "$cache_database_number" ; then
+        "$KUBECTL_BINARY" $kubeconfig_arg exec -t -n "default" "$cache_leader_pod_name" -- "/bin/bash" -c "redis-cli -a \$REDIS_PASSWORD -n $cache_database_number FLUSHDB"
+        cache_flush_status=$?
+        if [ $cache_flush_status -ne 0 ] ; then
+            echo "error encountered during attempt to clear portal persistence cache for portal $portal_id : $KUBECTL_BINARY returned status $cache_flush_status" >&2
+            #exit 1
```

Comment:
I had to comment out this line to ensure that we attempt to clear the cache for both clusters, even if the cache pod does not exist on one of them.